zebra-chain/src/block/height.rs

```diff
@@ -41,6 +43,54 @@ impl Height {
     pub const MAX_AS_U32: u32 = Self::MAX.0;
 }
 
+// Block heights live in a torsor, they can only be contructed but not computed directly.
+// Addition and substraction is done in the underlying types(u32) and not in the heights themselves.
+// This u32 numbers are a group and they can be added and substracted normally to get a new integer.
+// Having the result of the computation the `Height(X)` constructor can be used to build a new `Height`.
+
+// N = Height1.U - Height2.U
+// Given two `Height`s use the underlying value of them to compute the difference,
+// return this number as it is.
+impl Sub<Height> for Height {
+    type Output = Option<u32>;
+
+    fn sub(self, rhs: Height) -> Option<u32> {
+        self.0.checked_sub(rhs.0)
+    }
+}
+
+// H = Height(Height.U + N)
+// Given a `Height` and a number, sum the underlying value to the number,
+// with the result construct a new `Height` and return it.
+impl Add<u32> for Height {
+    type Output = Option<Height>;
+
+    fn add(self, rhs: u32) -> Option<Height> {
+        let result = self.0.checked_add(rhs);
+        match result {
+            Some(h) if (Height(h) <= Height::MAX) => Some(Height(h)),
+            Some(_) => None,
+            None => None,
```

Comment:
Simplification: instead of having two match arms that both return `None`, using a catch-all case `_` makes the logic clearer ("here is the success case, here is all other cases").

But this code might look a little different after changing from `u32` to `i32`.

Reply:
That was silly from my side, thanks! Changed it as part of https://github.com/ZcashFoundation/zebra/pull/1057/commits/4e3a77479985a2bce82a4cd9fb0d138ecda2fca4